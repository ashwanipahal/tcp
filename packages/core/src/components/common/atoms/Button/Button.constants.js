const BUTTON_VARIATION = {
  miniNav: 'mini-nav',
  mobileAppFilter: 'mobileApp-filter',
  mobileAppFilterIcon: 'mobileApp-filter-icon',
};

const BUTTON_FILL = {
  BLUE: 'BLUE',
  DARK: 'DARK',
  BLACK: 'BLACK',
};

export { BUTTON_VARIATION, BUTTON_FILL };
