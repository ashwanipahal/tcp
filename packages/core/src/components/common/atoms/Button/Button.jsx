import React from 'react';
import { PropTypes } from 'prop-types';
import styles from './Button.style';
import withStyles from '../../hoc/withStyles';

const Button = ({ children, className, ariaLabel, disabled, fullWidth }) => (
  <button
    disabled={disabled}
    aria-label={ariaLabel}
    className={className}
    type="button"
    fullWidth={fullWidth}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  ariaLabel: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  fullWidth: PropTypes.bool,
};

Button.defaultProps = {
  fullWidth: true,
};

export default withStyles(Button, styles);
export { Button as ButtonVanilla };
