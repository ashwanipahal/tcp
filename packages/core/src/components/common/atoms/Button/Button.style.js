import { css } from 'styled-components';

const ButtonStyles = css`
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  cursor: ${props => (props.disabled ? 'none' : 'pointer')};
  opacity: ${props => (props.disabled ? props.theme.opacity.medium : '1')};
  text-transform: uppercase;
  min-height: 42px;
  ${props =>
    props.buttonVariation === 'fixed-width'
      ? `
      min-width: 40px;
      background: ${props.theme.colors.BUTTON.NORMAL};
      color: ${props.theme.colors.BUTTON.TEXT};
      font-family: ${props.theme.fonts.secondaryFontSemilBoldFamily};
      font-size: ${props.theme.fonts.fontSize.button.size}px;
      font-weight: ${props.theme.fonts.fontWeight.semiBold};
      border: 1px solid ${props.theme.colors.BUTTON.BORDER};
      padding: 12px 20px;
      width: 100%;
    `
      : ''};
  ${props =>
    props.buttonVariation === 'variable-width'
      ? `
      min-width: 64px;
      background: ${props.theme.colors.BUTTON.NORMAL};
      color: ${props.theme.colors.BUTTON.TEXT};
      font-family: ${props.theme.fonts.secondaryFontFamily};
      font-size: ${props.theme.fonts.fontSize.button.size}px;
      font-weight: ${props.theme.fonts.fontWeight.black};
      border: 1px solid ${props.theme.colors.BUTTON.BORDER};
      padding: 12px 32px;
    `
      : ''};
  ${props =>
    props.fullWidth
      ? `
      width: 100% ;
    `
      : ''};

  &:hover {
    background: ${props => props.theme.colors.BUTTON.HOVER};
  }
  &:focus {
    background: ${props => props.theme.colors.BUTTON.FOCUS};
  }

  ${props =>
    props.customButton
      ? `
    background: #f3f3f3;
    color: #6a6a6a;
    border-radius: 10px;
    text-transform: uppercase;
    padding: 11px 0;
    border: 2px solid #d8d8d8;
    box-shadow: 10px 10px white inset, 12px 12px #f3f3f3;
    width: calc(100% - 14px);
    &:hover {
      background: #f2f9fe;
      box-shadow: 10px 10px white inset, 12px 12px #f2f9fe;
      border: 2px solid #9b9b9b;
    }
    `
      : ''}

  @media ${props => props.theme.mediaQuery.large} {
    ${props =>
      props.buttonVariation === 'fixed-width' ? 'min-height: 51px; padding: 16px 20px;' : ''};
    ${props =>
      props.buttonVariation === 'variable-width' ? 'min-height: 45px; padding: 16px 32px;' : ''};
  }
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default ButtonStyles;
