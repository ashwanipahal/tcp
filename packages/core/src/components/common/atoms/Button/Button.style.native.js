import { css } from 'styled-components/native';

const style = css`
  border: none;
  border-radius: 0;
  position: relative;
  opacity: ${props => (props.disabled ? props.theme.opacity.opacity.medium : '1')};
  text-transform: uppercase;
  min-height: 42px;
  text-align: center;
  ${props =>
    props.buttonVariation === 'fixed-width'
      ? `
    width: 40px;
    height: 100px;
    color: ${props.theme.colorPalette.black};
    font-size: ${props.theme.fonts.fontSize.button.size}px
    font-family: ${props.theme.typography.fonts.primary};
    font-weight: ${props.theme.typography.fontWeights.semibold};
    border: 1px solid ${props.theme.colorPalette.black};
    padding: 4px 20px;
    width: 100%;
    margin : 40px
  `
      : ''};

  ${props =>
    props.buttonVariation === 'variable-width'
      ? `
  width: ${props.width};
  height: ${props.width};
  min-width: 64px;
  background: ${props.theme.colorPalette.white};
  color: ${props.theme.colorPalette.black};
  font-family: ${props.theme.typography.fonts.secondary};
  font-size: ${props.theme.fonts.fontSize.button.size}px;
  font-weight: ${props.theme.typography.fontWeights.black};
  border: 1px solid ${props.theme.colorPalette.black};
  padding: 12px 32px;
  `
      : ''};

  ${props => (props.fullWidth ? ` width: 100% ; ` : '')};

  ${props =>
    props.customStyle === 'shadow-button'
      ? `
      background: ${props.theme.colors.PRIMARY.PALEGRAY};
      border-radius: 10px;
      padding: 4px 0;
      border: 2px solid ${props.theme.colors.PRIMARY.LIGHTGRAY};
      width: 100%;
      &:hover, &:focus {
        background: ${props.theme.colors.PRIMARY.COLOR1};
        box-shadow: 4px 4px white inset, 6px 6px ${props.theme.colors.PRIMARY.COLOR1};
        border: 2px solid ${props.theme.colors.PRIMARY.GRAY};
      }
      @media ${props.theme.mediaQuery.large} {
        padding: 11px 0;
        box-shadow: 10px 10px white inset, 12px 12px ${props.theme.colors.PRIMARY.PALEGRAY};
        width: calc(100% - 12px);
        &:hover, &:focus {
          box-shadow: 10px 10px white inset, 12px 12px ${props.theme.colors.PRIMARY.COLOR1};
        }
      }
    `
      : ''};
`;

export default style;
