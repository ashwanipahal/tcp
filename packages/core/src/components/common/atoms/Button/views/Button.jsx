import React from 'react';
import { PropTypes } from 'prop-types';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';

import styles from '../Button.style';

const Button = ({ children, className, ariaLabel, disabled, fullWidth }) => (
  <button
    disabled={disabled}
    aria-label={ariaLabel}
    className={className}
    type="button"
    fullWidth={fullWidth}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  ariaLabel: PropTypes.string,
  disabled: PropTypes.bool,
  fullWidth: PropTypes.bool,
};

Button.defaultProps = {
  fullWidth: true,
  ariaLabel: '',
  disabled: false,
};

export default withStyles(Button, styles);
export { Button as ButtonVanilla };
