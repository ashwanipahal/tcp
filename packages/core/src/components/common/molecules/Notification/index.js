export { default } from './views/Notification';
