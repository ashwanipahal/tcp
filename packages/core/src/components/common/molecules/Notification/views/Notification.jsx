import React from 'react';
import PropTypes from 'prop-types';
import { getIconPath } from '../../../../../utils';
import styles from '../Notification.style';
import withStyles from '../../../hoc/withStyles';
import BodyCopy from '../../../atoms/BodyCopy';

/**
 *
 * @param {String} className -  classname for the parent container
 * @param {String} status - to show error or success message
 * @param {String} message - error message
 * @param {node} children - notification child nodes.
 */
const Notification = ({ className, status, message, children }) => {
  const successIcon = getIconPath('circle-check-fill');
  const errorIcon = getIconPath('circle-error-fill');
  return (
    <div className={`${className} elem-pt-SM elem-pr-LRG elem-pb-SM elem-pl-LRG elem-mb-LRG`}>
      <img
        alt={status === 'success' ? 'success icon' : 'error icon'}
        src={status === 'success' ? successIcon : errorIcon}
        className="notification__image elem-mr-MED"
      />
      <BodyCopy fontSize="fs14" fontWeight="extrabold" fontFamily="secondary">
        {message}
        {children}
      </BodyCopy>
    </div>
  );
};

Notification.propTypes = {
  className: PropTypes.string,
  status: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  children: PropTypes.node,
};

Notification.defaultProps = {
  className: '',
  children: null,
};

export default withStyles(Notification, styles);
export { Notification as NotificationVanilla };
