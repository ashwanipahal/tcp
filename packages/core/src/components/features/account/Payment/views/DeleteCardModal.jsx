import React from 'react';
import withStyles from '../../../../common/hoc/withStyles';
import styles from '../styles/DeleteCardModal.style';
import Modal from '../../../../common/molecules/Modal';
import Notification from '../../../../common/molecules/Notification';
import GiftCardModalInfo from './GiftCardModalInfo.view';
import CreditCardModalInfo from './CreditCardModalInfo.view';
import VenmoCardModalInfo from './VenmoCardModalInfo.view';
import Button from '../../../../common/atoms/Button';
// @flow

type Props = {
  data: Object,
  className: string,
  openState: boolean,
  showUpdatedNotificationOnModal: boolean,
  labels: Object,
  setDeleteModalMountState: Function,
  onDeleteCard: Function,
};

/**
 * @function deleteCardModal The deleteCardModal component shows the address to delete.
 * This component includes the adress view, and confirm and cancel buttons
 * @param {data} data object with details to render in modal
 * @param {onDeleteAddress} onDeleteCards function to delete the address from the modal
 * @param {closeModalComponent} closeModalComponent function to close the modal
 * @param {className} className css to apply
 */
class DeleteCardModal extends React.Component<Props> {
  /**
   * @function onCloseModal  Used to render the JSX of the component
   * @param {closeModalComponent} closeModalComponent function to close the modal.
   * @return {[Function]} function called
   */
  onClose = () => {
    const { setDeleteModalMountState } = this.props;
    setDeleteModalMountState({ state: false });
  };

  /**
   * @function onCloseModal  Used to render the JSX of the component
   * @param {onDeleteCards} onDeleteCards function to delete the address.
   * @param {data} data object with details to render in modal
   * @return {[Function]} function called
   */
  onConfirm = () => {
    const { data, onDeleteCard } = this.props;
    const { description } = data;
    onDeleteCard({ creditCardId: description.creditCardId });
  };

  /**
   * @function renderModal  Used to render the JSX of the component
   * @param {modalType} modalType which modal to open : string - 'delete', 'verify'.
   * @param {data} data object with details to render in modal
   * @param {onDeleteAddress} onDeleteAddress function to delete the address from the modal
   * @param {closeModalComponent} closeModalComponent function to close the modal
   * @return {[Object]} JSX of the component
   */
  renderModal = () => {
    const { data, className, labels } = this.props;
    const ccType = data && data.description && data.description.ccType;
    const getAccNumbr = `${data.description.accountNo}`.slice(-4);
    const TotalExp = `${data.description.expMonth}/${data.description.expYear} `;
    const isCreditCard = ccType !== 'GiftCard' && ccType !== 'VENMO';
    const creditCardHeading = labels.ACC_LBL_MODAL_CREDIT_CARD_HEADING;
    const address = data.description.addressDetails ? data.description.addressDetails : null;
    return (
      <div className={className}>
        {isCreditCard && (
          <CreditCardModalInfo
            getAccNumbr={getAccNumbr}
            TotalExp={TotalExp}
            data={data}
            address={address}
            creditCardHeading={creditCardHeading}
            {...this.props}
            isCreditCard={isCreditCard}
          />
        )}
        {ccType === 'GiftCard' && (
          <GiftCardModalInfo
            getAccNumbr={getAccNumbr}
            TotalExp={TotalExp}
            data={data}
            {...this.props}
          />
        )}
        {ccType === 'VENMO' && (
          <VenmoCardModalInfo
            getAccNumbr={getAccNumbr}
            TotalExp={TotalExp}
            data={data}
            {...this.props}
          />
        )}
        <div className="deleteCardModal__btnWrapper">
          <Button
            buttonVariation="variable-width"
            fill="BLUE"
            onClick={this.onConfirm}
            className="deleteCardModal__deleteConfirm deleteCardModal__btn"
          >
            {data.buttons.confirm}
          </Button>
          <Button
            buttonVariation="variable-width"
            onClick={this.onClose}
            fill="RED"
            className="deleteCardModal__btn deleteCardModal__deleteCancel"
          >
            {data.buttons.cancel}
          </Button>
        </div>
      </div>
    );
  };

  /**
   * @function render  Used to render the JSX of the component
   * @param    {[Void]} function does not accept anything.
   * @return   {[Object]} JSX of the component
   */
  render() {
    const { className, data, openState, showUpdatedNotificationOnModal, labels } = this.props;
    const { heading } = data;
    return (
      <Modal
        fixedWidth
        isOpen={openState}
        onRequestClose={this.onClose}
        heading={heading}
        overlayClassName="TCPModal__Overlay"
        className={`TCPModal__Content, ${className}`}
        maxWidth="690px"
        minHeight="340px"
      >
        {showUpdatedNotificationOnModal && (
          <Notification
            status={showUpdatedNotificationOnModal}
            colSize={{ large: 11, medium: 7, small: 6 }}
            message={labels.ACC_LBL_ERROR_MSG}
          />
        )}

        {this.renderModal()}
      </Modal>
    );
  }
}

export default withStyles(DeleteCardModal, styles);
export { DeleteCardModal as DeleteCardModalVanilla };
