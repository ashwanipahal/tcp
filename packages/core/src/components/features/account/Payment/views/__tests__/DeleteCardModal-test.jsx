import { shallow } from 'enzyme';
import React from 'react';
import { DeleteCardModalVanilla } from '../DeleteCardModal';
import Button from '../../../../../common/atoms/Button';
import Notification from '../../../../../common/molecules/Notification';
// @flow

describe('Delete card Modal', () => {
  const data = {
    heading: 'DELETE ADDRESS',
    title: 'Are you sure you want to delete this address?',
    description: {
      firstName: 'test',
      lastName: 'test',
      addressLine: ['addressline 1', 'addressline 2'],
    },
    buttons: {
      cancel: 'No, Dont Cancel',
      confirm: 'Yes Delete',
    },
    cardText: {
      cardEnd: 'Card ending in ',
      expire: ' Expire on ',
    },
  };
  const labels = {
    paymentGC: {},
    common: {},
  };
  it('should render correctly', () => {
    const tree = shallow(<DeleteCardModalVanilla labels={labels} data={data} />);
    expect(tree).toMatchSnapshot();
    expect(tree.find(Button)).toHaveLength(2);
  });
  it('should call Cancel Button Correctly', () => {
    const mockedCloseModal = jest.fn();
    const tree = shallow(
      <DeleteCardModalVanilla
        data={data}
        labels={labels}
        setDeleteModalMountState={mockedCloseModal}
      />
    );
    tree
      .find(Button)
      .at(1)
      .simulate('click');
    expect(mockedCloseModal).toHaveBeenCalled();
  });
  it('should call confirm Button Correctly', () => {
    const mockedOnConfirm = jest.fn();
    const tree = shallow(
      <DeleteCardModalVanilla labels={labels} data={data} onDeleteCard={mockedOnConfirm} />
    );
    tree
      .find(Button)
      .at(0)
      .simulate('click');
    expect(mockedOnConfirm).toHaveBeenCalled();
  });
  it('should show notification on error', () => {
    const tree = shallow(
      <DeleteCardModalVanilla
        data={data}
        showUpdatedNotificationOnModal
        labels={{ errorMessage: 'error', paymentGC: {}, common: {} }}
      />
    );
    expect(tree.find(Notification)).toHaveLength(1);
  });
});
