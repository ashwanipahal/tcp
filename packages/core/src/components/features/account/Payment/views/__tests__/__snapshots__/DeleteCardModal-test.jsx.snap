// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete card Modal should render correctly 1`] = `
<errorBoundaryComponent
  className="TCPModal__Content, undefined"
  fixedWidth={true}
  heading="DELETE ADDRESS"
  maxWidth="690px"
  minHeight="340px"
  onRequestClose={[Function]}
  overlayClassName="TCPModal__Overlay"
>
  <div>
    <Styled(CreditCardModalInfo)
      TotalExp="undefined/undefined "
      address={null}
      data={
        Object {
          "buttons": Object {
            "cancel": "No, Dont Cancel",
            "confirm": "Yes Delete",
          },
          "cardText": Object {
            "cardEnd": "Card ending in ",
            "expire": " Expire on ",
          },
          "description": Object {
            "addressLine": Array [
              "addressline 1",
              "addressline 2",
            ],
            "firstName": "test",
            "lastName": "test",
          },
          "heading": "DELETE ADDRESS",
          "title": "Are you sure you want to delete this address?",
        }
      }
      getAccNumbr="ined"
      isCreditCard={true}
    />
    <div
      className="deleteCardModal__btnWrapper"
    >
      <Styled(Button)
        buttonVariation="variable-width"
        className="deleteCardModal__deleteConfirm deleteCardModal__btn"
        fill="BLUE"
        onClick={[Function]}
      >
        Yes Delete
      </Styled(Button)>
      <Styled(Button)
        buttonVariation="variable-width"
        className="deleteCardModal__btn deleteCardModal__deleteCancel"
        fill="RED"
        onClick={[Function]}
      >
        No, Dont Cancel
      </Styled(Button)>
    </div>
  </div>
</errorBoundaryComponent>
`;
